### DISCUSSION TITLE
Splitting generated files

### DISCUSSION THREAD

**Discussion by aminalaee** [koxudaxi/datamodel-code-generator#649] [Q&A] [answered]
Hi, this a great tool,

Is there a way to split output to different files?
I'm thinking something like what `openapi generator` does and create separate files for each class?
So each file will only contain a single class.

**Comment by koxudaxi** [comment, marked as answer]
You can define the model as modular if you add `.` in the model name.
There is an example.
https://github.com/koxudaxi/datamodel-code-generator/blob/master/tests/data/openapi/modular.yaml
https://github.com/koxudaxi/datamodel-code-generator/tree/master/tests/data/expected/main/main_modular
